### ISSUE TITLE
Implement the Photon checkboxes

### ISSUE THREAD

**Issue by ryanfeeley** [mozilla/fxa-content-server#6029]
Our checkboxes on newsletter signup, choose what to sync (and other places?) are out of date.

We should be using the new Photon style everywhere.

https://design.firefox.com/photon/components/checkboxes.html

child of #6083 
child of https://github.com/mozilla/fxa-content-server/issues/6261

**Comment by hritvi** [comment]
Can I work on this @ryanfeeley! Thanks

**Comment by vladikoff** [comment]
@hritvi sure!

**Comment by shane-tomlinson** [comment]
@MarcoZehe and I were able to chat FxA's a11y and he pointed out that the way we render checkboxes makes them unusable for screen readers since the original checkbox is hidden.

**Comment by ryanfeeley** [comment]
👍
